src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -68399,6 +68399,105 @@ static void F(object? x, object y)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.y").WithLocation(14, 9));
         }
 
+        [Fact]
+        [WorkItem(46206, "https://github.com/dotnet/roslyn/issues/46206")]
+        public void Tuple_Assignment_25()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(object x, string y)
+    {
+        (object, string?) t = new ValueTuple<object?, string>(item2: """", item1: null) { Item1 = x };
+        t.Item1.ToString();
+        t.Item2.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                    // (6,31): warning CS8619: Nullability of reference types in value of type '(object?, string)' doesn't match target type '(object, string?)'.
+                    //         (object, string?) t = new ValueTuple<object?, string>(item2: "", item1: null) { Item1 = x };
+                    Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new ValueTuple<object?, string>(item2: """", item1: null) { Item1 = x }").WithArguments("(object?, string)", "(object, string?)").WithLocation(6, 31));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(46206, "https://github.com/dotnet/roslyn/issues/46206")]
+        public void Tuple_Assignment_26()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(bool b, object x, object? y)
+    {
+        var t = new ValueTuple<object?, object, object?, object, object?, ValueTuple<object, object?>, object, ValueTuple<object, object?, object>>(
+            x,
+            x,
+            x,
+            b ? y : null!, // 1
```

Comment:
>null! [](start = 20, length = 5)

Consider using `x` here. The `null!` confused me.

Reply:
And elsewhere in this test where you use `null!`.

---
In reply to: [488872680](https://github.com/dotnet/roslyn/pull/47523#discussion_r488872680) [](ancestors = 488872680)

Reply:
The tests were just copied from existing tests for valuttuple FYI